docs/design/FeatureSwitches.md

```diff
@@ -32,3 +33,31 @@ When disabled, MAUI won't look for implicit cast operators when converting value
 If your library or your app defines an implicit operator on a type that can be used in one of the previous scenarios, you should define a custom `TypeConverter` for your type and attach it to the type using the `[TypeConverter(typeof(MyTypeConverter))]` attribute.
 
 _Note: Prefer using the `TypeConverterAttribute` as it can help the trimmer achieve better binary size in certain scenarios._
+
+## _MauiBindingInterceptorsSupport
+
+When enabled, MAUI will enable a source generator which will identify calls to the `SetBinding<TSource, TProperty>(this BindableObject target, BindableProperty property, Func<TSource, TProperty> getter, ...)` methods and generate optimized bindings based on the lambda expression passed as the `getter` parameter.
+
+This feature is a counterpart of [XAML Compiled bindings](https://learn.microsoft.com/dotnet/maui/fundamentals/data-binding/compiled-bindings).
+
+It is necessary to use this feature instead of the string-based bindings in NativeAOT apps and in apps with full trimming enabled.
+
+### Example use-case
+
+String-based binding in code:
+```c#
+label.BindingContext = new PageViewModel { Customer = new CustomerViewModel { Name = "John" } };
+label.SetBinding(Label.TextProperty, "Customer.Name");
+```
+
+Compiled binding in code:
+```c#
```

Comment:
I think you have to spell it out like this for GitHub to render syntax highlighting:
```suggestion
```csharp
```

Reply:
Both seem to work similarly "well"